task2/src/components/home/Main/HomeMain.jsx

```diff
@@ -1,21 +1,57 @@
+import { useState, useEffect } from 'react';
 import {
     Main,
-    BoldText,
 } from '@components/global';
 import {
     MovieCatalogFilter,
-    MovieCatalogCounter,
     MovieCatalog
-} from '@components/home'
+} from '@components/home';
+import { MovieService } from '@containers';
 
-const HomeMain = () => (
-    <Main>
-        <MovieCatalogFilter />
-        <MovieCatalogCounter>
-            <BoldText>9</BoldText><span> movies found</span>
-        </MovieCatalogCounter>
-        <MovieCatalog />
-    </Main>
-);
+//TODO: make right order when real API
+const HomeMain = () => {
+    const [movies, setMovies] = useState([]);
+    const [genres, setGenres] = useState([]);
+    //TODO: change state to const
+    const [activeFilter, setActiveFilter] = useState('ALL');
+    const [activeSort, setActiveSort] = useState('TITLE');
+    useEffect(async () => {
```

Comment:
@DrDiman, could you please help: is there some difference between using synchronous and `async` functions in `useEffect`? Which is better?

Reply:
in such case useEffect will return promise - and that's the problem.

u can use thenable chain to avoid this as 
```
    MovieService.getMovies()
    .then(d=> setMovies(d))
```
 check it here:
https://javascript.plainenglish.io/how-to-use-async-function-in-react-hook-useeffect-typescript-js-6204a788a435#:~:text=To%20do%20this%2C%20the%20function,compiler%20is%20yielding%20in%20Typescript.

Reply:
Thank you! I used an example from one website. I cannot find the link :(